modules/snooker/src/main/ts/ephox/snooker/model/TableGrid.ts

```diff
@@ -35,15 +35,17 @@ const findDiff = (xs: ElementNew[], comp: (a: SugarElement, b: SugarElement) =>
  *   rowspan: row span of the cell at (row, column)
  */
 const subgrid = (grid: RowCells[], row: number, column: number, comparator: (a: SugarElement, b: SugarElement) => boolean): { colspan: number; rowspan: number } => {
-  const restOfRow = getRow(grid, row).cells.slice(column);
-  const endColIndex = findDiff(restOfRow, comparator);
+  const gridRow = getRow(grid, row);
+  const isColRow = gridRow.section === 'colgroup';
+  const { rows, cols } = GridRow.extractGridDetails(grid);
+  const numColRows = cols.length;
 
-  const restOfColumn = getColumn(grid, column).slice(row);
-  const endRowIndex = findDiff(restOfColumn, comparator);
+  const colspan = findDiff(gridRow.cells.slice(column), comparator);
+  const rowspan = isColRow ? 1 : findDiff(getColumn(rows, column).slice(row - numColRows), comparator);
```

Comment:
Why was this changed from using `grid` to `rows`? As that doesn't look like it was needed and in fact complicates this since we need to then also adjust the row number passed. If there was a valid reason, can we please move the row number adjustment to a separate variable?

 Also is this even going to work if the `row` passed in is `0`, since the slice would then be `-1` potentially meaning it'll get everything except the last item?

Reply:
Re that last part, I guess it'd be doing `[].slice(-1)` which would still be an empty array. It also shouldn't really be able to get into that state since if you pass in `0` then `isColRow` would be true. However, I'd prefer if we weren't relying on side-effects like that for this to just happen to be working, as it can cause confusion/be hard to reason about.

Reply:
So after playing with this some more, I believe this should achieve the desired result here and also saves some cycles, since `getColumn` doesn't need to walk over every row.
```
const rowspan = isColRow ? 1 : findDiff(getColumn(grid.slice(row), column), comparator);
```

Reply:
I think I just overcomplicated it. I agree that the suggestion you have given should work properly.